ska_helpers/utils.py

```diff
@@ -0,0 +1,112 @@
+import functools
+
+__all__ = ['LazyDict']
+
+
+def _lazy_load_wrap(unbound_method):
+    @functools.wraps(unbound_method)
+    def wrapper(self, *args, **kwargs):
+        self._load()
+        return unbound_method(self, *args, **kwargs)
+    return wrapper
+
+
+class LazyVal:
+    """Value which is lazy-initialized using supplied function ``load_func``.
+
+    This class allows defining a module-level value that is expensive to
+    initialize, where the initialization is done lazily (only when actually
+    needed).
+
+    The lazy value is accessed using the ``get`` method.
+
+    Parameters
+    ----------
+    load_func : function
+        Reference to a function that returns a dict to init this dict object
+    *args
+        Arguments list for ``load_func``
+    **kwargs
+        Keyword arguments for ``load_func``
+    """
+    def __init__(self, load_func, *args, **kwargs):
+        self._load_func = load_func
+        self._args = args
+        self._kwargs = kwargs
+        self._loaded = False
+        super().__init__()
+
+    def get(self):
+        if not self._loaded:
+            self.val = self._load_func(*self._args, **self._kwargs)
+            self._loaded = True
+
+            # Delete these so pickling always works (pickling a func can fail)
+            del self._load_func
```

Comment:
I thought maybe there were some pickle-specific methods you could define in the class (short of a custom Pickler) that would be more general, but this is readable.

Reply:
I guess I was thinking of `__getstate__`.

Reply:
The current method would help with other serializations like YAML as well.